src/installer.ts

```diff
@@ -183,6 +218,11 @@ async function getInfoFromManifest(
     auth,
     'main'
   );
+
+  if (isLts(versionSpec)) {
+    versionSpec = findLtsVersionFromManifest(versionSpec, stable, releases);
```

Comment:
JFYI:

I tried to avoid fetching manifest multiple times, and incorporated the LTS resolving logic here. Since I set `check-latest = true` always, the resolving happens **before** toolcache code is executed, and the version is resolved at that point.

However, if we discuss and decide that `check-latest` should be automatically set to `true`, then I'll need to refactor code a little bit, and add additional manifest request before toolcache logic.

Reply:
That is the good question. Our general politic is using cache as much as possible. 

If customer specifies `14.x` as input, we will check local cache firstly. In my opinion, `lts/fermium` input should work exactly the same way like `14.x` because it means the same.

May be @konradpabjan @bryanmacfarlane have different opinion.

Reply:
That is the good question. 
Our general policy is using local cache as much as possible.

If user specifies version input as `14.x`, we will check local cache firstly. In my opinion, `lts/fermium` input should work exactly the same way like `14.x` (be full alias) so we don't need to set check-latest to true.

Reply:
Agree

For `setup-java` we set `chcek-latest=false`:

https://github.com/actions/setup-java/blob/ad01d131cc39a1e467480c19bf28c5a4fade0edc/action.yml#L23-L25

I think we should be consistent across the board. One of the main justifications was to help speed up workflows. If the tool-cache is preferred than the majority of workflows will be ever so slightly faster (and that reduces COGS and infrastructure costs for us 😃 ). If something is not in the tool-cache than the action will reach out to externally try to find it. Sometimes there might be delays before the cache is updated with the latest version but i think it's minimal in the broader context of things. If someone absolutely needs the the latest versions always they can set `check-latest=true` themselves but in terms of optimizing for the most common use-case, I think `check-latest` should be `false`.

Reply:
I see, makes perfect sense.
My initial idea was to have closer behavior with `nvm`.
E.g. when you do `nvm use 14` it will use **any** existing 14th node on your machine.
But when you do `nvm use lts/fermium` it will always check `nodejs` website, and will use the **latest** 14th node.

But in the context of GitHub Action it's a little bit different. There is no "local machine", and basically on every run an action installs the latest possible node (whether it be from toolcache or from node website).

> Sometimes there might be delays before the cache is updated with the latest version but i think it's minimal in the broader context of things.

👆 I agree, then it's fine to use toolcache, and not set `check-latest`.
Especially considering this is a general policy. 👍 

**TL;DR** I initially thought within the bound of `nvm`, my oversight. In the context of GitHub Actions it makes sense to not set `check-latest=true`. I'll need to refactor the code a little bit (to fetch manifest separately), so please don't review yet. 😅

Reply:
@maxim-lobanov @konradpabjan Remastered the whole logic. ✔️ 

Bonus: added `manifest` local variable, so if there was manifest fetched for `lts` check, there won't be another manifest fetch.

However, I didn't touch the existing (for smaller diff, easier review, and lower regression chance).
That means first call to `resolveVersionFromManifest` doesn't "cache" `manifest` as before, thus the second call to `getInfoFromManifest` will fetch `manifest` again. Hope that makes sense.